### ISSUE TITLE
Issues with steam_api64 not getting into Mac build

### ISSUE THREAD

**Issue by gwachnitz** [rlabrecque/Steamworks.NET#426]
HI, I have been having an issue recently where I get:
```
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./steam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./steam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./steam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./steam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./steam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./steam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/steam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.dylib
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.so
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/./libsteam_api64.bundle
Fallback handler could not load library /Users/gwachnitz/Library/Application Support/Steam/steamapps/common/Rogue Empire/Mac.app/Contents/Frameworks/MonoEmbedRuntime/osx/libsteam_api64
System.DllNotFoundException: steam_api64
  at (wrapper managed-to-native) Steamworks.NativeMethods:SteamAPI_RestartAppIfNecessary (Steamworks.AppId_t)
  at Steamworks.SteamAPI.RestartAppIfNecessary (AppId_t unOwnAppID) [0x00000] in <filename unknown>:0 
  at SteamManager.Awake ()
```

`

I am really baffled by this. Can I manually add the libs somewhere? If so where would I find them.

I did saw similar posts but they don't seem to work :S

Using Unity 2017

**Comment by gwachnitz** [comment]
Compatibility for macos its returning false for some reason:

```
if (plugin.GetCompatibleWithPlatform(platform) == enable) {
			return false;
		}
```

**Comment by gwachnitz** [comment]
Still no clue what is happening. Had to disable steam integration for Mac users.

**Comment by yaakov-h** [comment]
The native library name should only be `steam_api64` for 64-bit Windows. On macOS it should be just `steam_api` which is a universal ("fat") binary containing multiple architecture slices.

**Comment by gwachnitz** [comment]
> The native library name should only be `steam_api64` for 64-bit Windows. On macOS it should be just `steam_api` which is a universal ("fat") binary containing multiple architecture slices.

OK, but why does the mac build try to access this lib?

**Comment by gwachnitz** [comment]
This is strange, every time I tick the mac box after  building my game it unticks automatically. 

![image](https://user-images.githubusercontent.com/17207401/123275386-a8824200-d4da-11eb-8210-3fdd0bdb81c0.png)

**Comment by gwachnitz** [comment]
Any ideas?

**Comment by rlabrecque** [comment]
This should be completely resolved in 20.0.0+ now that we ship .meta files for the plugins!

**Comment by gwachnitz** [comment]
Just wanted to clarify that for some reason updating to 20.0.0+ didn't cut it for me. I had to rename the steam_api to steam_api64 for the mac lib
